src/Registry/TypeRegistry.php

```diff
@@ -815,7 +810,10 @@ public function prepare_type( string $type_name, $config ) {
 					$prepared_type = new WPEnumType( $config );
 					break;
 				case 'input':
-					$prepared_type = new WPInputObjectType( $config, $this );
+					// Cast config to exact type expected
+					/** @var array{name?: string|null, description?: string|null, fields: (callable(): iterable<string, mixed>)|iterable<string, mixed>, parseValue?: callable(array<string, mixed>): mixed, astNode?: \GraphQL\Language\AST\InputObjectTypeDefinitionNode|null, extensionASTNodes?: array<\GraphQL\Language\AST\InputObjectTypeExtensionNode>|null} $input_config */
```

Comment:
Why are all these type coercions necessary (everywhere in this PR).

At bare minimum we should `@phpstan-import-type`, but if we need coercion to begin with its usually a sign of a bigger issue.

This makes me very nervous in a breaking refactor.

I'll be taking a pass on this after #3288

Reply:
Ya, this could be some of my ignorance with phpstan. There was a lot of phpstan warnings I don't think I fully had a grasp on and was working through things to satisy it. Open to other ways of handling it

Reply:
Mostly resolved by #3289